[socket.c] Enable AES for SSL by default

black-dragon74 opened the pull request:
`/proc/cpuinfo` file is only present on Linux hosts which leads to AES
cipher not being enabled for SSL on other hosts despite the CPU
supporting the hw accel.

This approach uses the assembly instruction `cpuid` to check for AES bit
presence, making it a global solution on *NIX hosts. Perks; we don't have
to deal with files which helps with performance and efficiency.

Change-Id: Ide9ebc0a39da2b02c71d7f766f6b0473433b8b8a
Signed-off-by: black-dragon74 <niryadav@redhat.com>

mykaul commented:
Did you manage to find a processor that does NOT have acceleration for AES? It's been in the market for >10 years. I think we should unconditionally enable it.

black-dragon74 commented:
> Did you manage to find a processor that does NOT have acceleration for AES? It's been in the market for >10 years. I think we should unconditionally enable it.

No I did not, it has been around since Westmere/Jaguar family, so highly unlikely. With ref to this PR, I just refactored the existing code to ditch fops.

Guess I'll just add it to `DEFAULT_CIPHER_LIST` and remove all the code related to checks. WDYT?

Regards

mykaul commented:
> > Did you manage to find a processor that does NOT have acceleration for AES? It's been in the market for >10 years. I think we should unconditionally enable it.
> 
> No I did not, it has been around since Westmere/Jaguar family, so highly unlikely. With ref to this PR, I just refactored the existing code to ditch fops.
> 
> Guess I'll just add it to `DEFAULT_CIPHER_LIST` and remove all the code related to checks. WDYT?

IMHO, it should be fine to set it as default in all cases.

xhernandez commented:
I'm ok with this change, but you need to update the commit message to explain what's really done. It doesn't match with the code.

xhernandez commented:
/run regression

black-dragon74 committed:
[socket.c] Enable AES cipher for SSL by default

Almost all of the CPUs in the market have HW Accel for AES enabled by default.
So we can skip the check for it and enable it by default.

Signed-off-by: black.dragon74 <nickk.2974@gmail.com>

gluster-ant commented:
CLANG-FORMAT FAILURE:
Before merging the patch, this diff needs to be considered for passing clang-format

```diff --git a/rpc/rpc-transport/socket/src/socket.c b/rpc/rpc-transport/socket/src/socket.c
index 9100f58f1..ef71b45d9 100644
--- a/rpc/rpc-transport/socket/src/socket.c
+++ b/rpc/rpc-transport/socket/src/socket.c
@@ -56,7 +56,8 @@
 #if !defined(DEFAULT_VERIFY_DEPTH)
 #define DEFAULT_VERIFY_DEPTH 1
 #endif
-#define DEFAULT_CIPHER_LIST "AES128:EECDH:EDH:HIGH:!3DES:!RC4:!DES:!MD5:!aNULL:!eNULL"
+#define DEFAULT_CIPHER_LIST                                                    \
+    "AES128:EECDH:EDH:HIGH:!3DES:!RC4:!DES:!MD5:!aNULL:!eNULL"
 #define DEFAULT_DH_PARAM SSL_CERT_PATH "/dhparam.pem"
 #define DEFAULT_EC_CURVE "prime256v1"
 
```

xhernandez commented:
/run regression

black-dragon74 commented:
Commenting to have this patch merged as well.

Regards

CC: @amarts @xhernandez
